runners/spark/src/main/java/org/apache/beam/runners/spark/SparkRunner.java

```diff
@@ -218,11 +185,12 @@ public void run() {
       final JavaSparkContext jsc = SparkContextFactory.getSparkContext(mOptions);
       final EvaluationContext evaluationContext = new EvaluationContext(jsc, pipeline);
 
+      initAccumulators(mOptions, jsc);
```

Comment:
In streaming mode, seems like we call `initAccumulators(...)` both here and [here](https://github.com/apache/beam/pull/2171/files#diff-b68cfd0ea98c566fbe6024c78e89342cR83). 
While multiple calls cause no harm in this case, do we really need both?

I assume the latter is required to satisfy [this](https://github.com/apache/beam/pull/2171/files#diff-08c9c68e4e7d25b7acf16d32fc66d601R136), i.e., to make sure things have been initialised by the time the pipeline is translated and needs an `AggregatorsAccumulator` instance.

I think we can make `SparkRunner#run()`'s interaction with `SparkRunnerStreamingContextFactory` and the pipeline translation in particular, clearer (pipeline translation is hidden inside `SparkRunnerStreamingContextFactory#create()` which IMHO is not ideal).

Anyhow, this is out of the scope of this PR, but could be nice to look into sometime.